### PR TITLE
Firmware CR-NL

### PR DESCRIPTION
**Pull request by stffrdhrn**
Support different combinations of CR/NL as is now down in the litex bios.

### CODE REVIEW

File: firmware/ci.c [hunk 1]
```diff
@@ -231,6 +231,7 @@ static char *readstr(void)
 	char c[2];
 	static char s[128]; // Needs to fit a full mode line ~100 chars.
 	static int ptr = 0;
+	static char skip = 0;
```
Comment (mithro) [inline firmware/ci.c]: Should this technically be `'\0'` ?

Comment (mithro) [inline firmware/ci.c]: Maybe something like `skip_if_next` is a better name? 
We only skip the value if it is exactly the *next* value to be received right? IE `\ni\r` wouldn't hit the skip?

File: firmware/ci.c [hunk 2]
```diff
@@ -263,6 +264,14 @@ static char *readstr(void)
 		if(readchar_nonblock()) {
 			c[0] = readchar();
 			c[1] = 0;
+
+			/* Check if we have anything to skip.  */
```
Comment (mithro) [inline firmware/ci.c]: I don't really like the reusage of `skip` variable here...

Maybe
```c
bool should_skip = (c[0] == skip);
skip = '\0';
if (should_skip)
  return NULL;
```

or

```c
if (c[0] == skip) {
  skip = '\0';
  return NULL;
} else {
  skip = '\0';
}
```

**Comment by mithro** [comment]
Concept seems fine! Just some minor nits around the style....

File: firmware/ci.c (hunk 1 above)
Comment (stffrdhrn) [reply firmware/ci.c]: Yes, I was just following the style.  I can update other places they are assigning char to `0` to `'\0'`

Comment (stffrdhrn) [reply firmware/ci.c]: Yes, only next.  I can change the name something like that.  Again, I was just following the naming from litex.

File: firmware/ci.c (hunk 2 above)
Comment (stffrdhrn) [reply firmware/ci.c]: I had it like the second case before.  I was just trying to be clever. I will revise.

**Comment by stffrdhrn** [comment]
I will clean them up.  The CI failure doesn't look related, I couldn't see what it was in the log though.

**Comment by mithro** [comment]
LGTM!

**Pull request merged**